### ISSUE TITLE
Yet another Lint/FormatParameterMismatch false positive related to string interpolation

### ISSUE THREAD

**Issue by zor-el** [rubocop/rubocop#12672]
A similar issue (#11521) appears to have been addressed, but it didn't seem to cover this type of case:

```ruby
# frozen_string_literal: true

prefix = 'alert'
message = 'some message'
puts format('%10s: %s', prefix, message)
padding = 10
puts format("%#{padding}s: %s", prefix, message)
```

The `format(...)` invocations are equivalent, yet the latter causes an offense:

```
... W: Lint/FormatParameterMismatch: Format string is invalid because formatting sequence types (numbered, named or unnumbered) are mixed.
puts format("%#{padding}s: %s", prefix, message)
     ^^^^^^
```

## Expected behavior

No Lint/FormatParameterMismatch offense reported.

## Actual behavior

See above

## Steps to reproduce the problem

See above

## RuboCop version

Tested with:

```
1.60.2 (using Parser 3.3.0.5, rubocop-ast 1.30.0, running on ruby 3.2.2) [x86_64-darwin21]
1.57.2 (using Parser 3.3.0.5, rubocop-ast 1.30.0, running on ruby 3.2.2) [x86_64-darwin21]
1.60.2 (using Parser 3.3.0.5, rubocop-ast 1.30.0, running on ruby 2.7.8) [x86_64-darwin21]
```

**Comment by pboling** [comment]
I am also experiencing this issue:
```
sprintf("| %-#{key_offset}s | %-#{val_offset}s |", key, value)
```
RuboCop says:
```
    Lint/FormatParameterMismatch: Number of arguments (2) to `sprintf` doesn't match the number of fields (1).
```
Version: 1.57.2 - old I know, but since this is an open issue, I'm posting it to show another example of the failure, which is probably the same regardless of version.
